### PR TITLE
fix: ensure app dirs exist 

### PR DESCRIPTION
**Pull request by njhale**
Ensure the WORKSPACE_DIR and SCRIPTS_PATH directories are created before attempting to use them and refactor some misc related code.

### CODE REVIEW

File: actions/scripts/new.tsx [hunk 1]
```diff
@@ -4,18 +4,17 @@ import { SCRIPTS_PATH } from '@/config/env';
 import fs from 'fs/promises';
 
 export async function newFile(name: string, instructions: string, fileName: string) {
-    try {
-        const files = await fs.readdir(SCRIPTS_PATH());
-        const gptFiles = files.filter(file => file.endsWith('.gpt'));
+    const scriptsPath = SCRIPTS_PATH()
+    await fs.mkdir(scriptsPath, {recursive: true})
```
Comment (njhale) [inline actions/scripts/new.tsx:8]: Note: `fs.mkdir` doesn't throw an error if the directory already exists when the `recursive: true` option is passed.

File: actions/threads.tsx [hunk 2]
```diff
@@ -129,15 +129,3 @@ export async function renameThread(id: string, name: string) {
...
-    const threadsDir = THREADS_DIR();
-    const threadPath = path.join(threadsDir,id);
-
-    if (thread.state) await fs.writeFile(path.join(threadPath, STATE_FILE), thread.state);
-    if (thread.meta) {
-        const existingMeta = await fs.readFile(path.join(threadPath, META_FILE), "utf-8");
-        const mergedMeta = { ...JSON.parse(existingMeta), ...thread.meta };
-        await fs.writeFile(path.join(threadPath, META_FILE), JSON.stringify(mergedMeta));
-    }
-}
```
Comment (njhale) [inline actions/threads.tsx:143]: Seems like this isn't used

File: app/api/file/[name]/[tool]/route.ts [hunk 3]
```diff
@@ -1,4 +1,6 @@
+
 export const dynamic = 'force-dynamic' // defaults to autover'
+import {NextResponse} from "next/server";
```
Comment (njhale) [inline app/api/file/[name]/[tool]/route.ts:2]: My linter was complaining about `Response.json` not existing here.

Comment (tylerslaton) [reply app/api/file/[name]/[tool]/route.ts:2]: Fair, it is auto imported I believe. Also this whole API section can get removed soonish. It isn't used right now, not important but just adjacent.

**Review by tylerslaton — APPROVED** [review]
Looks good.

File: actions/threads.tsx (hunk 2 above)
Comment (tylerslaton) [reply actions/threads.tsx:143]: It isn't right now.

**Pull request merged**